app/http.php

```diff
@@ -36,11 +44,53 @@
         // 'document_root' => __DIR__.'/../public',
         // 'enable_static_handler' => true,
         'http_compression' => true,
+        'dispatch_func' => 'dispatch',
         'http_compression_level' => 6,
         'package_max_length' => $payloadSize,
         'buffer_output_size' => $payloadSize,
     ]);
 
+function dispatch(Server $server, $fd, $type, $data = null)
+{
+    global $workerNumber, $domains;
+
+    $safeThreadsPercent = intval(App::getEnv('_APP_SAFE_THREADS_PERCENT', 80)) / 100;
+
+    $safeThreads = (int) floor($workerNumber * $safeThreadsPercent);
+
+    $risky = false;
+    if ($data && str_contains($data, 'POST') && str_contains($data, '/executions')) {
```

Comment:
$data also includes body, right? I am worried there will be false positive.
Can we split and only get first line for this check? First line should only have method and URL. Then we can also do checks for starts_with and ends_with.

Reply:
We can, but I also worry about the performance?

Reply:
We can benchmark, split with limit 2 should be very fast - after first split its quick. First split is after method and path, which shouldtn be too much, max ~2k chars.

I would love to avoid this.. Scarry to have a bug that causes random timeouts when there is specific word inside a body of my request

Reply:
I looked into it, splitting by \n itself requires the code to scan through the entire $data.. I don't see a value trying to parse it because the compute we save by removing str_contains is taken up in splitting. Plus, the check the host header requires us to find the host header among all the split lines.

Reply:
I dont think full scan is needed if we limit the split,third parameter of https://www.php.net/manual/en/function.explode.php

If we tell it to split only into 2 parts by the first occurrence, I don't want to believe PHP will continue looking for others. We can benchmark it to double check.